core/src/scene/importer.cpp

```diff
@@ -82,141 +69,112 @@ Node Importer::applySceneImports(const std::string& scenePath, const std::string
         if (import.IsScalar()) {
-            import = normalizePath(import.Scalar(), parentPath);
+            import = Url(import.Scalar()).resolved(base).string();
         } else if (import.IsSequence()) {
-            for (Node imp : import) {
-                if (imp.IsScalar()) { imp = normalizePath(imp.Scalar(), parentPath); }
+            for (Node path : import) {
+                if (path.IsScalar()) {
+                    path = Url(path.Scalar()).resolved(base).string();
+                }
             }
         }
     }
-}
 
-void Importer::normalizeSceneDataSources(Node &root, const std::string &parentPath) {
+    // Resolve data source URLs.
+
     if (Node sources = root["sources"]) {
         for (auto source : sources) {
             if (Node sourceUrl = source.second["url"]) {
-                sourceUrl = normalizePath(sourceUrl.Scalar(), parentPath);
+                sourceUrl = Url(sourceUrl.Scalar()).resolved(base).string();
             }
         }
     }
-}
-
-void Importer::setNormalizedTexture(Node& texture, const std::vector<std::string>& names,
-                                    const std::string& parentPath) {
-
-    for (size_t index = 0; index < names.size(); index++) {
-
-        auto& name = names[index];
-        if (isBase64Data(name)) {
-            continue;
-        }
-
-        std::string normTexPath;
-
-        // if texture url is a named texture then move on (this has been already resolved
-        if (m_globalTextures.find(name) != m_globalTextures.end()) { continue; }
-
-        // get normalized texture path
-        if (m_textureNames.find(name) == m_textureNames.end()) {
-            normTexPath = normalizePath(name, parentPath);
-            m_textureNames[name] = normTexPath;
-        } else {
-            normTexPath = m_textureNames.at(name);
-        }
-
-        // set yaml node with normalized texture path
-        if (names.size() > 1) {
-            texture[index] = normTexPath;
-        } else {
-            texture = normTexPath;
-        }
-    }
-}
 
-void Importer::normalizeSceneTextures(Node& root, const std::string& parentPath) {
+    // Resolve global texture URLs.
 
-    if (Node textures = root["textures"]) {
-        for (auto texture : textures) {
-            if (Node textureUrl = texture.second["url"]) {
-                if (!isBase64Data(textureUrl.Scalar())) {
-                    setNormalizedTexture(textureUrl, {textureUrl.Scalar()}, parentPath);
+    if (Node texturesNode = root["textures"]) {
+        for (auto texture : texturesNode) {
+            if (Node textureUrlNode = texture.second["url"]) {
+                if (textureUrlNode.IsScalar()) {
+                    textureUrlNode = Url(textureUrlNode.Scalar()).resolved(base).string();
                 }
                 m_globalTextures.insert(texture.first.Scalar());
             }
         }
     }
 
+    // Resolve inline texture URLs.
+
     if (Node styles = root["styles"]) {
```

Comment:
No - probably only the m_globalTextures for the imports up to this point should be considered